catkin_tools/argument_parsing.py

```diff
@@ -201,32 +241,75 @@ def handle_make_arguments(input_make_args, force_single_threaded_when_running_te
         if run_tests:
             wide_log('Forcing "-j1" for running unit tests.')
-            make_args.append('-j1')
-
-    # If no -j/--jobs/-l/--load-average flags are in make_args
-    if not extract_jobs_flags(' '.join(make_args)):
-        # If -j/--jobs/-l/--load-average are in MAKEFLAGS
-        if 'MAKEFLAGS' in os.environ and extract_jobs_flags(os.environ['MAKEFLAGS']):
-            # Do not extend make arguments, let MAKEFLAGS set things
-            pass
-        else:
-            # Else extend the make_arguments to include some jobs flags
-            # Use the number of CPU cores
-            try:
-                jobs = cpu_count()
-                make_args.append('-j{0}'.format(jobs))
-                make_args.append('-l{0}'.format(jobs))
-            except NotImplementedError:
-                # If the number of cores cannot be determined, do not extend args
-                pass
+            jobs_dict['jobs'] = 1
+
+    if 'jobs' in jobs_dict:
+        make_args.append('-j{0}'.format(jobs_dict['jobs']))
+    if 'load-average' in jobs_dict:
+        make_args.append('-l{0}'.format(jobs_dict['load-average']))
+
     return make_args
 
 
+def configure_make_args(make_args, internal_make_jobserver):
+    """Initialize the internal GNU Make jobserver or configure it as a pass-through
+
+    :param make_args: arguments to be passed to GNU Make
+    :type make_args: list
+    :param internal_make_jobserver: if true, use the internal jobserver
+    :type make_args: bool
+    :rtype: tuple
+    """
+
+    # Configure default jobs options: use all CPUs in each package
+    try:
+        # NOTE: this will yeild greater than 100% CPU utilization
+        n_cpus = cpu_count()
+        jobs_flags = {
+            'jobs': n_cpus,
+            'load-average': n_cpus}
+    except NotImplementedError:
+        # If the number of cores cannot be determined, limit to one job
+        jobs_flags = {
+            'jobs': 1,
+            'load-average': 1}
+
+    # Get MAKEFLAGS from environment
+    makeflags_jobs_flags = extract_jobs_flags(os.environ.get('MAKEFLAGS', '')) or []
+    using_makeflags_jobs_flags = len(makeflags_jobs_flags) > 0
+    jobs_flags.update(extract_jobs_flags_values(''.join(makeflags_jobs_flags)))
+
+    # Extract make jobs flags (these override MAKEFLAGS)
+    cli_jobs_flags = extract_jobs_flags(''.join(make_args)) or []
+    using_cli_flags = len(cli_jobs_flags) > 0
+    jobs_flags.update(extract_jobs_flags_values(''.join(cli_jobs_flags)))
+    if cli_jobs_flags:
+        # Remove jobs flags from cli args if they're present
+        make_args = re.sub(cli_jobs_flags, '', ' '.join(make_args)).split()
+
+    # Instantiate a jobserver
+    jobserver = MakeJobServer.get_instance(
+        jobs_flags.get('jobs', None),
+        internal_make_jobserver)
+
+    # If the jobserver is supported and enabled, override the jobs flags
+    using_jobserver = jobserver.supported and internal_make_jobserver
```

Comment:
Yeah, there's no reason to check both, this is just slop. I'd actually like it if `supported` didn't keep having to be checked.